tests/conftest.py

```diff
@@ -104,11 +105,13 @@ def use_new_resolver(request):
     """Set environment variable to make pip default to the new resolver.
     """
     new_resolver = request.config.getoption("--new-resolver")
+    features = set(os.environ.get("PIP_USE_FEATURE", "").split())
```

Comment:
This is what I imagine after applying this patch:

| Original PIP_USE_FEATURE | With --new-resolver | Without --new-resolver |
| ------------------------ | ------------------- | ---------------------- |
|                          | 2020-resolver       |                        |
| 2020-resolver            | 2020-resolver       |                        |
| 2020-resolver spam       | 2020-resolver spam  | spam                   |
| spam                     | 2020-resolver spam  | spam                   |

IIUC are you suggesting the following, regardless of original PIP_USE_FEATURE, by *explicitly setting via a CLI option*?

| With --new-resolver | Without --new-resolver |
| ------------------- | ---------------------- |
| 2020-resolver       |                        |

I thought of this earlier, but it doesn't seem to work well when we have more features to toggle.

Reply:
For both `2020-resolver` and `fast-dep`, I’d argue we should actually *unset* what ever the user sets to run the tests, to ensure we’re testing the right thing. So the correct approach would be to overwrite this environment variable in one fixture.

Also, this is not an issue if the tests are run with tox as suggested in the development guide, since tox does not pass through environment variables by default.

Reply:
As tox doesn't pass env vars, can we use PIP_USE_FEATURE as the common state, e.g. if there's another fixture use_spam, is this PR's current behavior OK for testing 4 cases when they are on/off?

Reply:
Yes, I believe the intention when `use_new_resolver` was original implemented was exactly this; the environment variable is used as the main state for what feature(s) the test is currently testing against.